### PR TITLE
bison: Declare m4 dependency correctly

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/bison/package.py [hunk 1]
```diff
@@ -35,7 +35,8 @@ class Bison(AutotoolsPackage):
 
     version('3.0.4', 'a586e11cd4aff49c3ff6d3b6a4c9ccf8')
 
-    depends_on('m4', type='build')
+    # This is a run-time dependency as well
+    depends_on('m4')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/bison/package.py]: Same here, probably run+build

Comment (eschnett) [inline var/spack/repos/builtin/packages/bison/package.py]: Isn't `build+run` the default?

Comment (davydden) [inline var/spack/repos/builtin/packages/bison/package.py]: Don't thinks so, check the docs, I comment from cell

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bison/package.py]: Default is `build, link`. See [documentation](http://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-specs)

Comment (davydden) [inline var/spack/repos/builtin/packages/bison/package.py]: Quick way to check: Spack info bison

Comment (eschnett) [inline var/spack/repos/builtin/packages/bison/package.py]: Done.

**Pull request merged**